tests/mock_network/mod.rs

```diff
@@ -547,3 +550,57 @@ fn pause_node_and_poll(env: &Environment, nodes: &mut Vec<TestNode>) -> PausedSt
 
     state
 }
+
+#[test]
+fn neighbour_update() {
+    // 1. Create two sections, A and B.
+    // 2. Change the set of elders of B.
+    // 3. Send a message from B to A to make A request an update from B.
+    // 4. Verify A's view of B is up to date.
+
+    let env = Environment::new(NetworkParams {
+        elder_size: LOWERED_ELDER_SIZE,
+        recommended_section_size: LOWERED_ELDER_SIZE + 1,
+    });
+    let mut rng = env.new_rng();
+    let mut nodes = create_connected_nodes_until_split(&env, &[1, 1]);
+
+    let prefix_a = Prefix::default().pushed(rng.gen());
+    let prefix_b = prefix_a.sibling();
+
+    // A's view of B is initially up to date.
+    assert!(section_knowledge_is_up_to_date(
+        &nodes,
+        &prefix_a,
+        &prefix_b,
+        env.elder_size()
+    ));
+
+    // Remove at most elder_size - 1 elders, so section A still knows at least one elder from B and
+    // so can still contact them.
+    let num_elders_to_remove = rng.gen_range(1, LOWERED_ELDER_SIZE);
+    info!(
+        "Removing {} elders from {:?}",
+        num_elders_to_remove, prefix_b
+    );
+
+    // Add nodes that will replace the removed elders.
+    for _ in 0..num_elders_to_remove {
+        add_node_to_section(&env, &mut nodes, &prefix_b);
+    }
+    poll_until(&env, &mut nodes, |nodes| {
+        all_nodes_joined(nodes, 0..nodes.len())
+    });
+
+    // Remove some elders from B.
+    for _ in 0..num_elders_to_remove {
+        let name = remove_elder_from_section(&mut nodes, &prefix_b);
+        poll_until(&env, &mut nodes, |nodes| node_left(nodes, &name));
+    }
+
+    // Send a message from B to A to trigger the update request.
```

Comment:
shall we have a check before sending the message that the knowledge is now out of date ?

Reply:
We could, but I don't think we should. Although the knowledge is out of date at this point, we don't require it to be. What we require it only that the knowledge is eventually updated after the message is sent.

Reply:
> Although the knowledge is out of date at this point, we don't require it to be

So you mean the knowledge could already be updated (due to some other messaging flow). And the extra message is just a `safe-net` to guarantee that update happens, right?

Reply:
It could in theory, but there is no other message flow in this case. My point was that we don't really care about the knowledge being out of date before the message. We only care about it being updated afterwards. And I think we should assert only things we care about. 

That said, adding the assertion would slightly change the meaning of the test as it would be about explicitly requiring that it's the message that causes the knowledge to get updated and not something else. There can be some value in that, I'm not sure. If you think it's worthwhile, I'll add it there.

Reply:
> That said, adding the assertion would slightly change the meaning of the test as it would be about explicitly requiring that it's the message that causes the knowledge to get updated and not something else. There can be some value in that, I'm not sure. If you think it's worthwhile, I'll add it there.

Yes, I agree adding an assertion changes the meaning slightly as you say. However, if so, we may need to know what that extra cause the update? So far, as I understand, it shall only be the message flow causing the update?

Reply:
OK, will add the assert.